src/machine/board_teensy36.go

```diff
@@ -0,0 +1,87 @@
+// +build nxp,mk66f18,teensy36
+
+package machine
+
+import (
+	"device/nxp"
+)
+
+// CPUFrequency returns the frequency of the ARM core clock (180MHz)
+func CPUFrequency() uint32 { return 180000000 }
+
+// ClockFrequency returns the frequency of the external oscillator (16MHz)
+func ClockFrequency() uint32 { return 16000000 }
```

Comment:
We don't use anything like that elsewhere in TinyGo, and don't think it should be added unless there is a real need for it (to avoid complicating the API).

Reply:
`sleepWait` configures the low power timer to use the external reference clock for timing. This means that my math is wrong, because `cyclesPerMilli` and `cyclesPerMicro` are calculated from the CPU clock. `sleepWait` should be using values calculated from `ClockFrequency()` instead of `CPUFrequency()`. The low power timer does have the option of using the 'internal reference clock', MCGIRCLK, but I'm not sure what that is.

Reply:
Yes and no. The way I calculate the countdown for the timer is incorrect and results in sleeps 180MHz/16MHz = 11 times shorter than they should be. However, I use `ticks()` to determine when I'm done sleeping, so `sleepTicks` still blocks for the correct duration.

Given that the frequency of the external clock is board-specific, I put it in the board file, so other boards would be able to override it.

BTW I said `sleepWait` in my previous comment when I meant `sleepTicks`. I've edited it.

Reply:
To directly answer, yes there is something that I implemented correctly, and to fix it I need to know the frequency of the external oscillator. Since that is board-specific, I didn't want to hard code it. Do you have a suggestion for resolving this? I could use an entirely different timer, but the low power timer seemed like the best option for scheduler sleeps.